### PR TITLE
[nexus3] Use postStart container lifecycle hook to run configure.sh

### LINKED ISSUE #124: Consider moving configure.sh into a postStart lifecycle command
Hi there,

One of the disadvantages of executing the configure.sh, given that it mostly executes in the background, as one of the commands is that if it fails, there is no feedback to helm for it to be able to recognise the failure. I've experimented with a post start lifecycle command (see below), and it works well. If the command fails, helm is able to recognise the failure and roll it back. Is it worth considering to replace both the `command` and `args` keys (plus removing the & from configure.sh) for the nexus3 container?

```
          lifecycle:
            postStart:
              exec:
                command:
                  - "sh"
                  - "-c"
                  - "${SONATYPE_DIR}/nexus/conf/configure.sh"
```

Thanks for this chart.

Regards,

### PR DESCRIPTION
**Pull request by grumpydude**
The following changes are included in this PR.

* Replaces the container command and args with a postStart lifecycle
* Removes the backgrounding of the outer loop in configure.sh 
* Fixes the order of configure.sh where the roles were still created prior to the repos.
* Updates comments in the anonymous user setup of configure.sh that mentioned metrics.
* Adds a default value for root_password in configure.sh in order to pass lint tests.

Closes #124

### CODE REVIEW

File: charts/nexus3/templates/deployment.yaml [hunk 1]
```diff
@@ -74,15 +74,13 @@ spec:
...
-              bash /opt/sonatype/nexus/conf/configure.sh;
-              {{- end }}
-              /opt/sonatype/start-nexus-repository-manager.sh;
+          lifecycle:
+            postStart:
+              exec:
+                command:
+                  - "sh"
+                  - "-c"
+                  - "${SONATYPE_DIR}/nexus/conf/configure.sh"
```
Comment (stevehipwell) [inline charts/nexus3/templates/deployment.yaml:83]: I think this should still be wrapped in the `{{- if .Values.config.enabled }}` staement.

File: charts/nexus3/files/configure.sh [hunk 2]
```diff
@@ -149,40 +149,10 @@ do
...
-
-      echo "Role configured."
-    fi
-  done
-
   json_file="${base_dir}/conf/anonymous-user.json"
   if [ -f "${json_file}" ]
   then
-    echo "Configuring metrics..."
+    echo "Configuring anonymous user..."
```
Comment (stevehipwell) [inline charts/nexus3/files/configure.sh:155]: The anonymous user is an implementation detail of metrics which is why it wasn't previously referenced, I'd be happier if this was changed to `echo "Configuring anonymous user for metrics..."` (with the same logic for the completed message).

**Review by stevehipwell — COMMENTED** [review]
Thanks for this PR, good spot on the `root_password` not being checked correctly. I'm not sure how the move of the role in configure was missed, I'm going to have to put it down to incompetence on my part! if you're happy to do the suggested minor changes I will get this merged ASAP.

One further question, which version of K8s have you tested the lifecycle hook on?

File: charts/nexus3/templates/deployment.yaml (hunk 1 above)
Comment (grumpydude) [reply charts/nexus3/templates/deployment.yaml:83]: Yes, that's correct, and I can fix that.

File: charts/nexus3/files/configure.sh (hunk 2 above)
Comment (grumpydude) [reply charts/nexus3/files/configure.sh:155]: Ok, will do.

**Comment by grumpydude** [comment]
Note that this has now been tested  successfully on v1.16.15, v1.17.12, v1.18.9 and v1.19.3.

File: charts/nexus3/templates/deployment.yaml (hunk 1 above)
Comment (grumpydude) [reply charts/nexus3/templates/deployment.yaml:84]: Updated.

File: charts/nexus3/files/configure.sh (hunk 2 above)
Comment (grumpydude) [reply charts/nexus3/files/configure.sh]: Updated.

**Comment by stevehipwell** [comment]
Thanks @grumpydude!

**Pull request merged**